subprojects/android-test/test-inhouse-runner/src/main/kotlin/com/avito/android/runner/TestRunEnvironment.kt

```diff
@@ -183,19 +183,24 @@ private fun parseSentryConfig(argumentsProvider: ArgsProvider): SentryConfig {
     }
 }
 
-private fun parseStatsDConfig(argumentsProvider: ArgsProvider): StatsDConfig {
+internal fun parseStatsDConfig(argumentsProvider: ArgsProvider): StatsDConfig {
     val host = argumentsProvider.getOptionalArgument("statsDHost")
-    val port = argumentsProvider.getOptionalIntArgument("statsDPort")
```

Comment:
Why we didn't fail here?

Reply:
The code was:

```
 override fun getOptionalIntArgument(name: String): Int? {
        val defaultValue = Int.MIN_VALUE
        val result = bundle.getInt(name, defaultValue)
        return if (result == defaultValue) {
            null
        } else {
            result
        }
    }
```

We just returned null and fell to `Disabled`

Reply:
Yeah, I see the code. I meant that we should fail but not return defaults.
I propose to use defaults only for absent values.
We can check it by `containsKey` before reading.

Reply:
you mean introduce boolean like "useStatsd" and then required params if true?

Reply:
> introduce boolean like "useStatsd" and then required params if true
Yes, it will be easier than juggling multiple optional arguments.

Reply:
There is another option. 
We can check it even for an optional argument, but only if a bundle contains the key.